Content.Client/_RMC14/Xenonids/Watch/XenoWatchBui.cs

```diff
@@ -71,15 +246,131 @@ private void OnSearchBarChanged(LineEditEventArgs args)
         if (_window is not { Disposed: false })
             return;
 
+        SearchBarText = args.Text;
+
         foreach (var child in _window.XenoContainer.Children)
         {
+
             if (child is not XenoChoiceControl control)
                 continue;
 
             if (string.IsNullOrWhiteSpace(args.Text))
+            {
                 control.Visible = true;
+                UsingSearchBar = false;
+                UpdateList();
+            }
             else
-                control.Visible = control.NameLabel.GetMessage()?.Contains(args.Text, StringComparison.OrdinalIgnoreCase) ?? false;
+            {
+                UsingSearchBar = true;
+                control.Visible =
+                    control.NameLabel.GetMessage()?.Contains(args.Text, StringComparison.OrdinalIgnoreCase) ?? false;
+            }
+        }
+    }
+
+    private void OnButtonToggled(BaseButton.ButtonEventArgs args)
+    {
+        string name = args.Button.Name ?? string.Empty; // button name is emtpy, what i need is the text
+        if (name == string.Empty)
+        {
+            return;
+        }
+        if (ShownXenos.ContainsKey(name))
+        {
+            ShownXenos[name] = !ShownXenos[name];
+            args.Button.ModulateSelfOverride = UpdateButtonColor(ShownXenos[name]);
+        }
+        ClearSearchBar();
+        UpdateList();
+    }
+
+
+    private void OnTierToggled(BaseButton.ButtonEventArgs args)
+    {
+        args.Button.ModulateSelfOverride = UpdateButtonColor(args.Button.Pressed);
+        if (args.Button.Name != null && int.TryParse(args.Button.Name.Split()[1], out var number)) // get the tier number frmo the button name
+        {
+            foreach (var xeno in tiers[number])
+            {
+                ShownXenos[xeno.Name.Replace(" ","")] = args.Button.Pressed;
+            }
+        }
+        UpdateTierChilds();
+        ClearSearchBar();
+        UpdateList();
+    }
+
+    private void UpdateTierChilds()
+    {
+        if (_window is not { Disposed: false })
+            return;
+        foreach (var child in _window.RowContainer.Children)
+        {
+            if (child is not XenoTierRow control)
+                continue;
+            foreach (var xenocountchild in control.XenosContainer.Children) //i love nested loops (theres probably a better way to do this)
```

Comment:
```suggestion
            foreach (var xenocountchild in control.XenosContainer.Children)
```
Please, take a look https://github.com/0tito/RMC-14/pull/3/files#diff-a2e816d103e251264a7e1472861fc962ac2a63751ef04aa80dbb8fc374f8ae95R295